### PR TITLE
 pkp/pkp-lib#10511 fix and cleanup invalid series navigation menu items

### LINKED ISSUE #10511: [OMP] Fatal error when creating a series menu item without content
**Describe the bug**
After creating a series menu item without linking it to any item (see image below), a fatal error happens.

image

```php
PHP Fatal error:  Uncaught TypeError: PKP\section\Repository::get(): Argument #1 ($id) must be of type int, string given, called in classes/services/NavigationMenuService.php on line 157 and defined in lib/pkp/classes/section/Repository.php:58
Stack trace:
#0 classes/services/NavigationMenuService.php(157): PKP\section\Repository->get()
#1 [internal function]: APP\services\NavigationMenuService->getDisplayStatusCallback()
#2 lib/pkp/classes/plugins/Hook.php(139): call_user_func_array()
#3 lib/pkp/classes/plugins/Hook.php(113): PKP\plugins\Hook::run()
#4 lib/pkp/classes/services/PKPNavigationMenuService.php(382): PKP\plugins\Hook::call()
```

The strange thing that needs to be analyzed is that a call to `$navigationMenuItem->getPath()` outputs Laravel code `�*�escapeWhenCastingToString`.

A workaround must be left for bad cases, and the field should be probably turned into required. If the ID is a physical field, it can be setup as not null, but a further script should clear invalid data on the upgrade.

It's also needed to check if the bug applies to OMP 3.3

**To Reproduce**
1. Create a series menu item, like in the sample image
2. Add it to the navigation
3. See the error

**What application are you using?**
OMP 3.4

**Pull requests**
- OMP 3.4: https://github.com/pkp/omp/pull/1763
- OMP main: https://github.com/pkp/omp/pull/1772
- OMP 3.5 (postgres upgrade error fix): https://github.com/pkp/omp/pull/1875
- OMP main (postgres upgrade error fix): https://github.com/pkp/omp/pull/1882

### PR DESCRIPTION
**Pull request by kaitlinnewson**

### CODE REVIEW

**Review by jonasraoni — CHANGES_REQUESTED** [review]
Hi @kaitlinnewson! I just requested one update.

**Comment by jonasraoni** [comment]
Ah, I've spotted another problem...

The offending menu item can be the parent of other menu items, so if you remove it, the dependent items will be gone as well... In fact not, because a foreign key was forgotten (I'll create/update another issue to cover it), anyway, it needs to be addressed 🙃

So again, we have two options:
- Removing such item should cause its children to take its place on the tree.
- We setup the `path` with `null` and "convert" such items to a placeholder (it doesn't exist, so the closest to it would be a URL item pointing perhaps to `./`).

**Comment by jonasraoni** [comment]
FYI the current behavior is to hide everything which is under an orphan menu item, which I don't think it's ideal 🤔

File: classes/migration/upgrade/v3_5_0/I10511_RemoveSeriesMenuItems.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+class I10511_RemoveSeriesMenuItems extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        DB::table('navigation_menu_items')
+            ->where('type', 'NMI_TYPE_SERIES')
+            ->where('path', '�*�escapeWhenCastingToString')
```
[the remark this answers is not in the document]
Comment (kaitlinnewson) [reply classes/migration/upgrade/v3_5_0/I10511_RemoveSeriesMenuItems.php:30]: Reviewing this again - @jonasraoni do you mean to do a join, and then remove all entries that do not belong to an existing series?

[the remark this answers is not in the document]
Comment (jonasraoni) [reply classes/migration/upgrade/v3_5_0/I10511_RemoveSeriesMenuItems.php:30]: Yep, there's just an extra detail that I wrote below https://github.com/pkp/omp/pull/1772#issuecomment-2497880502 :)

**Comment by kaitlinnewson** [comment]
@jonasraoni PR has been updated and ready for another look. I noticed there is only one level of nesting in the menus, so I opted to set the parent to `null` instead, which keeps it assigned in the navigation menu. Alternatively we could delete it from `navigation_menu_item_assignments` and then it would still exist but just not be assigned to the menu.

File: classes/migration/upgrade/v3_5_0/I10511_RemoveSeriesMenuItems.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    {
+        $invalidNavigationMenuItemIds = DB::table('navigation_menu_items')
+            ->where('type', 'NMI_TYPE_SERIES')
+            ->whereNotIn('path', function (Builder $query) {
+                $query->select('series_id')->from('series');
+            })->pluck('navigation_menu_item_id');
+
+        DB::table('navigation_menu_item_assignments')
+            ->whereIn('parent_id', $invalidNavigationMenuItemIds)
+            ->update(['parent_id' => null]);
```
Comment (jonasraoni) [inline classes/migration/upgrade/v3_5_0/I10511_RemoveSeriesMenuItems.php:37]:
```suggestion
        if (!$invalidNavigationMenuItemIds->count()) {
            return;
        }

        DB::table('navigation_menu_item_assignments')
            ->whereIn('parent_id', $invalidNavigationMenuItemIds)
            ->update(['parent_id' => null]);
```

**Review by jonasraoni — APPROVED** [review]
It would be better to do both operations without materializing the IDs, but even on a large multi-journal installation it's unlikely that we'll have many defective items 🙃

**Pull request merged**